src/crimson/osd/watch.cc

```diff
@@ -81,9 +81,11 @@ seastar::future<> Watch::connect(crimson::net::ConnectionRef conn, bool)
 {
   if (this->conn == conn) {
     logger().debug("conn={} already connected", conn);
+    return seastar::now();
+  }
+  if (timeout_timer.armed()) {
```

Comment:
nit, `timeout_timer.cancel()` is a no-op if the timer is not armed.

Reply:
Thanks!